### PR TITLE
add sync status logic

### PR DESCRIPTION
**Pull request by Shamwayne**
Add Verify sync status logic

### CODE REVIEW

File: hightouch_blueprints/verify_sync.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import argparse
+import sys
+import requests
+import shipyard_utils as shipyard
+import errors
```
Comment (ericelsken) [inline hightouch_blueprints/verify_sync.py]: Same comment on relative import if possible.

**Review by ericelsken — CHANGES_REQUESTED** [review]

File: hightouch_blueprints/verify_sync.py (hunk 1 above)
Comment (blakeburch) [reply hightouch_blueprints/verify_sync.py]: And same comment as I made on the other PR. :)

File: hightouch_blueprints/verify_sync.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+def handle_sync_run_data(sync_run_data):
+    """
+    Analyses sync run data to determine status and print sync run information
+    
+    Returns:
+        status_code: Exit Status code detailing sync status
+    """
+    run_id = sync_run_data['id']
+    status = sync_run_data['status']
```
Comment (blakeburch) [inline hightouch_blueprints/verify_sync.py:84]: All of the sync_run_data references are wrong. Should be `sync_run_data['data'][0]['field_to_pull_in]`

Comment (blakeburch) [reply hightouch_blueprints/verify_sync.py:84]: Or you could just limit the initial storage of `sync_run_data` to be `sync_run_data['data'][0]` so these would all be valid.

File: hightouch_blueprints/verify_sync.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        if sync_status_code == requests.codes.ok:
+            return sync_run_json
+            
+        elif sync_status_code == 400: # Bad request
+            print("Sync status request failed due to Bad Request Error.")
+            sys.exit(errors.EXIT_CODE_BAD_REQUEST)
+            
+        elif sync_status_code == 401: # Incorrect credentials 
+            print("Incorrect credentials, check your access token")
+            sys.exit(errors.EXIT_CODE_INVALID_CREDENTIALS)
```
Comment (blakeburch) [inline hightouch_blueprints/verify_sync.py:57]: Need a handling for wrong sync id. But this apparently just returns a successful response but with `{'data': []}`.

File: hightouch_blueprints/verify_sync.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            print(f"Check status Validation failed: {sync_run_json['details']}")
+            sys.exit(errors.EXIT_CODE_BAD_REQUEST)
+        
+        else:
+            print("Unknown Error when sending request")
+            sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
+        
+    except Exception as e:
+        print(f"Sync status for Sync {sync_id}:{sync_run_id} failed due to: {e}")
+        sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
```
Comment (blakeburch) [inline hightouch_blueprints/verify_sync.py:73]: Same issue as the other PR. Status codes aren't being properly assessed and it jumps straight to the exception with a message of `Expecting value: line 1 column 1 (char 0)`

File: hightouch_blueprints/errors.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+EXIT_CODE_FINAL_STATUS_SUCCESS = 0
+EXIT_CODE_INVALID_CREDENTIALS = 200
+EXIT_CODE_BAD_REQUEST = 201
+EXIT_CODE_UNKNOWN_ERROR = 202
+EXIT_CODE_SYNC_REFRESH_ERROR = 210
+EXIT_CODE_FINAL_STATUS_INVALID = 211
```
Comment (blakeburch) [inline hightouch_blueprints/errors.py]: What is this? it's not used anywhere

File: hightouch_blueprints/verify_sync.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            f"Completed at: {sync_run_data['finishedAt']}"
+        )
+        status_code = errors.EXIT_CODE_FINAL_STATUS_SUCCESS
+        
+    elif sync_run_data['finishedAt'] is None:
+        print(
+            f"Sync run {run_id} still Running. ",
+            f"Current records processed: {sync_run_data['records_processed']}"
+        )
+        status_code = errors.EXIT_CODE_STATUS_RUNNING
```
Comment (blakeburch) [inline hightouch_blueprints/verify_sync.py]: This exit code doesn't exist on errors.py

File: hightouch_blueprints/verify_sync.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    
+    elif status == "failed":
+        error_info = sync_run_data['error']
+        print(f"Sync run {run_id} failed. {error_info}")
+        status_code = errors.EXIT_CODE_STATUS_FAILED
```
Comment (blakeburch) [inline hightouch_blueprints/verify_sync.py]: This exit code doesn't exist in `errors.py`

File: hightouch_blueprints/verify_sync.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        status_code = errors.EXIT_CODE_FINAL_STATUS_SUCCESS
+        
+    elif sync_run_data['finishedAt'] is None:
+        print(
+            f"Sync run {run_id} still Running. ",
+            f"Current records processed: {sync_run_data['records_processed']}"
+        )
+        status_code = errors.EXIT_CODE_STATUS_RUNNING
+    
+    elif status == "failed":
```
Comment (blakeburch) [inline hightouch_blueprints/verify_sync.py:99]: Is `success` and `failed` an exhaustive list of final statuses for hightouch? I couldn't find a list, but I'm assuming you did. Would love to document it.

**Review by blakeburch — CHANGES_REQUESTED** [review]

File: hightouch_blueprints/verify_sync.py (hunk 4 above)
Comment (Shamwayne) [reply hightouch_blueprints/verify_sync.py:73]: It took a bit to realize, but i'd placed the response.json() outside the success function. Not all error responses return a json response, some return plain text or just the status code and nothing else. 

I've gone through both rudderstack and hightouch to see if there were any other instances were i did this and fixed them

File: hightouch_blueprints/verify_sync.py (hunk 8 above)
Comment (Shamwayne) [reply hightouch_blueprints/verify_sync.py:99]: Unfortunately no. Those are what i've found through trial and error

**Review by blakeburch — APPROVED** [review]

**Pull request merged**